File: program_ir.py
from typing import List
from symbol_ir import SymbolIR
from scope_ir import ScopeIR
from import_ir import ImportIR
from binding_ir import BindingIR


class ProgramIR:
    def __init__(self, module_name: str, file_path: str, scopes: List[ScopeIR], symbols: List[SymbolIR], imports: List[ImportIR], decls: List[BindingIR]):
        self.module_name=module_name
        self.file_path=file_path
        self.scopes=scopes
        self.symbols=symbols
        self.imports=imports
        self.decls=decls

File: rust/src/ir/nodes/program_ir.rs
use crate::ir::nodes::{decl_ir::DeclIR, import_ir::ImportIR, scope_ir::ScopeIR, symbol_ir::SymbolIR};

#[derive(Debug, Clone)]
pub struct ProgramIR {
    pub module_name: String,
    pub file_path: String,
    pub scopes: Vec<ScopeIR>,
    pub symbols: Vec<SymbolIR>,
    pub imports: Vec<ImportIR>,
    pub decls: Vec<DeclIR>,
}
